python/pyspark/sql/tests/test_udtf.py

```diff
@@ -1031,6 +1031,16 @@ def eval(self, row: Row):
             [Row(a=6), Row(a=7)],
         )
 
+    def test_df_argument(self):
+        class TestUDTF:
+            def eval(self, row: Row):
+                if row["id"] > 5:
+                    yield row["id"],
+
+        func = udtf(TestUDTF, returnType="a: int")
+        df = self.spark.range(8)
+        assertDataFrameEqual(func(df.argument()), [Row(a=6), Row(a=7)])
```

Comment:
Shall we compare with the result from SQL, which clarifies the equivalent SQL and also checks the schema.

Reply:
Sounds good. I will combine test_udtf_with_table_argument_query and test_df_argument once Spark Connect support is ready.